2011-10-20  Carlos O'Donell  <carlos@systemhalted.org>
 
	* ports/sysdeps/hppa/stackinfo.h: Include elf.h and define 
	DEFAULT_STACK_PERMS.
 
---
 ports/sysdeps/hppa/stackinfo.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/ports/sysdeps/hppa/stackinfo.h
+++ b/ports/sysdeps/hppa/stackinfo.h
@@ -22,6 +22,12 @@
 #ifndef _STACKINFO_H
 #define _STACKINFO_H	1
 
+#include <elf.h>
+
+/* Default to an executable stack.  PF_X can be overridden if PT_GNU_STACK is
+ * present, but it is presumed absent.  */
+#define DEFAULT_STACK_PERMS (PF_R|PF_W|PF_X)
+
 /* On PA the stack grows up.  */
 #define _STACK_GROWS_UP	1
 
